handlers/upload.go

```diff
@@ -103,24 +124,23 @@ func (d *CatalystAPIHandlersCollection) handleUploadVOD(w http.ResponseWriter, r
 	var requestID = config.RandomTrailer(8)
 	log.AddContext(requestID, "source", uploadVODRequest.Url)
 
-	// find source segment URL
-	var tURL string
-	for _, o := range uploadVODRequest.OutputLocations {
-		if o.Outputs.SourceSegments {
-			tURL = o.URL
-			break
-		}
+	// Create a separate subdirectory for the source segments
+	// Use the output directory specified in request as the output directory of transcoded renditions
+	targetURL, err := uploadVODRequest.getTargetURL()
+	if err != nil {
+		return false, errors.WriteHTTPBadRequest(w, "Invalid request payload", err)
 	}
-	if tURL == "" {
-		return false, errors.WriteHTTPBadRequest(w, "Invalid request payload", fmt.Errorf("no source segment URL in request"))
+	// Hack for web3.storage to distinguish different jobs, before calling Publish()
+	// Can be removed after we address this issue: https://github.com/livepeer/go-tools/issues/16
+	if targetURL.Scheme == "w3s" {
+		targetURL.Host = requestID
```

Comment:
Could this be sth like `log.AddContext(requestID, "w3s-url", targetURL.String())`?

Reply:
Good idea, added.